File: backend/src/main/java/com/vvise/demo/repository/UserRepository.java
package com.vvise.demo.repository;

import com.vvise.demo.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByEmail(String email);
    Optional<User> findByProviderAndProviderId(User.AuthProvider provider, String providerId);
    boolean existsByEmail(String email);
}

File: backend/src/main/kotlin/com/vvise/demo/repository/UserRepository.kt
package com.vvise.demo.repository

import com.vvise.demo.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByEmail(email: String): User?
    fun findByProviderAndProviderId(provider: User.AuthProvider, providerId: String): User?
    fun existsByEmail(email: String): Boolean
}
